packages/web/src/pages/search-page-v2/RecentSearches.tsx

```diff
@@ -0,0 +1,318 @@
+import { useCollectionCoverArt2 } from 'hooks/useCollectionCoverArt'
+import { useMedia } from 'hooks/useMedia'
+import { useTrackCoverArt2 } from 'hooks/useTrackCoverArt'
+import { useProfilePicture } from 'hooks/useUserProfilePicture'
+import { profilePage } from 'utils/route'
+
+const MAX_RECENT_SEARCHES = 12
+
+const { removeItem, clearHistory } = searchActions
+const { getUserId } = accountSelectors
+const { getSearchHistory } = searchSelectors
+
+const messages = {
+  album: 'Album',
+  clear: 'Clear Recent Searches',
+  goTo: 'Go to: ',
+  playlist: 'Playlist',
+  profile: 'Profile',
+  remove: 'Remove recent search',
+  title: 'Recent searches',
+  track: 'Track'
+}
+
+const RecentSearchSkeleton = () => (
+  <Flex
+    w='100%'
+    pv='s'
+    ph='xl'
+    css={{
+      justifyContent: 'space-between'
+    }}
+  >
+    <Flex gap='m'>
+      <Skeleton w='40px' h='40px' />
+
+      <Flex direction='column' gap='s'>
+        <Skeleton w='120px' h='12px' />
+        <Skeleton w='100px' h='12px' />
+      </Flex>
+    </Flex>
+  </Flex>
+)
+
+type RecentSearchProps = {
+  children: React.ReactNode
+  linkTo: string
+  searchItem: SearchItem
+  title: string
+}
+
+const RecentSearch = (props: RecentSearchProps) => {
+  const { children, linkTo, searchItem, title } = props
+  const { color } = useTheme()
+  const dispatch = useDispatch()
+
+  const handleClickRemove = useCallback<MouseEventHandler>(
+    (e) => {
+      e.stopPropagation()
+      e.preventDefault()
+      dispatch(removeItem({ searchItem }))
+    },
+    [dispatch, searchItem]
+  )
+
+  return (
+    <Link to={linkTo}>
+      <Flex
+        w='100%'
+        pv='s'
+        ph='xl'
+        css={{
+          justifyContent: 'space-between',
+          cursor: 'pointer',
+          ':hover': {
+            backgroundColor: color.background.surface2
+          }
+        }}
+        role='button'
+        aria-label={`${messages.goTo} ${title}`}
+      >
+        <Flex gap='m'>{children}</Flex>
+        <IconButton
+          aria-label={messages.remove}
+          icon={IconClose}
+          color='subdued'
+          size='s'
+          onClick={handleClickRemove}
+        />
+      </Flex>
+    </Link>
+  )
+}
+
+const RecentSearchTrack = (props: { searchItem: SearchItem }) => {
+  const { searchItem } = props
+  const { id } = searchItem
+  const { data: track, status } = useGetTrackById({ id })
+
+  const image = useTrackCoverArt2(track?.track_id, SquareSizes.SIZE_150_BY_150)
+
+  if (status === Status.LOADING) return <RecentSearchSkeleton />
+
+  if (!track) return null
+  const { permalink, title, user } = track
+
+  if (!user) return null
+
+  return (
+    <RecentSearch searchItem={searchItem} title={title} linkTo={permalink}>
+      <Artwork src={image} w='40px' borderRadius='xs' />
+      <Flex direction='column' css={{ alignItems: 'flex-start' }}>
+        <Text variant='body' size='s'>
+          {title}
+        </Text>
+        <Flex alignItems='baseline'>
```

Comment:
can this all be a "text" component, set the variants, and then have UserLink inside it?

Reply:
Oh yes that'll be much better, I had to do think jankiness bc UserNameAndBadges doesn't use harmony and I couldn't override the styles